pkg/apis/extensions/types.go

```diff
@@ -228,6 +228,28 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
+}
+
+// DeploymentRollback stores the information required to rollback a deployment.
+type DeploymentRollback struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Required: This must match the Name of a deployment.
+	Name string `json:"name"`
+	// This must match the Namespace of a deployment. An empty namespace is
+	// equivalent to the "default" namespace.
+	Namespace string `json:"namespace,omitempty"`
+	// The annotations to be updated to a deployment
+	UpdatedAnnotations map[string]string `json:"updatedAnnotations,omitempty"`
+	// The config of this deployment rollback.
+	Rollback *RollbackConfig `json:"rollback"`
```

Comment:
`rollbackTo` seems fine and plays together with `revision`: `rollbackTo.revision`